web/src/engine/main/src/keymanEngine.ts

```diff
@@ -253,10 +255,27 @@ export default class KeymanEngine<
           this.core.resetContext(target);
         }
       },
-      predictionContext: new PredictionContext(this.core.languageProcessor, this.core.keyboardProcessor),
+      predictionContext: predictionContext,
       keyboardCache: this.keyboardRequisitioner.cache
     });
 
+    /*
+     * Handler for post-processing once a suggestion has been applied.
+     *
+     * This is called after the suggestion is applied but _before_ new
+     * predictions are requested based on the resulting context.
+     */
+    this.core.languageProcessor.on('suggestionapplied', () => {
```

Comment:
On a completely separate note, I just got to thinking about that `core` field on the base engine object.

Just a mild brainstorm, but...

- `keyman.processing` - currently known as `keyman.core`
- `keyman.processing.predictions` - currently known as `keyman.core.languageProcessor`
- `keyman.processing.jsKeystrokes` - currently known as `keyman.core.keyboardProcessor`

With potential future entries:
- `keyman.processing.ldmlKeystrokes` (as private)
- `keyman.processing.jsKeystrokes` -> private
- `keyman.processing.keystrokes` - as a common interface for `js-processor` and `ldml-processor`

I'm not sure this is outright the _best_ nomenclature, but it's probably an improvement over what we currently have.

Any thoughts, @mcdurdin?

Reply:
I don't think that we should be renaming too many things at this point -- the move from common to web is a big enough change.

That said, I'm also really not keen on the new names. I feel like `keyman.core` is a better and simpler abstraction than `keyman.processing`. And `languageProcessor` is also a better abstraction than `predictions`.

But this is a yak shave I don't think we should be getting into at this point.

Reply:
Ah, okay.  I'd figure use of `core` there would bother you, since Web is "not Core" and doesn't quite match it, either.

Reply:
`core` is there because of an old decision but I don't know that we need to change that now -- it is what it is, and change for the sake of change will have other onflow costs.